### PR TITLE
Fresh 1114

### PR DESCRIPTION
**Pull request by metas-rc**

### CODE REVIEW

File: de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java [hunk 1]
```diff
@@ -426,4 +427,19 @@ private void dynamicDisplay(final int fieldIndex)
 		// Ask the model to create the parameters
 		return model.createProcessInfoParameters();
 	}
+
+	/**
+	 * #782 Request focus on the first process parameter (if possible)
+	 */
+	public void focustFirstParameter()
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java]: typo?

File: de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java [hunk 2]
```diff
@@ -426,4 +427,19 @@ private void dynamicDisplay(final int fieldIndex)
...
+	{
+		if(fieldEditors.size() < 1)
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java]: why not using isEmpty() in this case?

File: de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java [hunk 3]
```diff
@@ -426,4 +427,19 @@ private void dynamicDisplay(final int fieldIndex)
...
+		{
+			// there are no parameters in this process. Nothing to focus
+			return;
+		}
+		final VEditor editor = fieldEditors.get(0);
+		getComponent(editor).requestFocusInWindow();
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java]: what happens if requestFocusInWindow() returns false.
Instead, i would suggest to iterate the editors, requestFOcusInWindow() and stop on first one which returns true.

File: de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessPanel.java [hunk 4]
```diff
@@ -185,19 +187,38 @@ public Dimension getPreferredSize()
...
+
+	/**
+	 * #782: Focus on the first parameter of the process when the process panel is opened, so it can be directly changed from keyboard
+	 */
+	private final WindowListener onWindowOpenFocusFirstParameter = new WindowAdapter()
+	{
+		@Override
+		public void windowOpened(final WindowEvent e)
+		{
+			if(parameterPanel == null)
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessPanel.java]: also, i would suggest to request focus only if CARDNAME_ProcessParameters.equals(getCurrentCardName()).

I would also extract the code from here and have ProcessPanel.requestFocusInWindow() method. Here, just call it.

File: de.metas.adempiere.adempiere/client/src/main/java-legacy/org/compiere/grid/ed/VLookup.java [hunk 5]
```diff
@@ -596,104 +596,125 @@ public void setReadWrite (final boolean value)
...
 			editorComp.requestFocus();
 		}
-	}	//	requestFocus
+	}	// requestFocus
+
+	/**
+	 * #782 Request Focus In Window
+	 */
+	@Override
+	public boolean requestFocusInWindow()
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/client/src/main/java-legacy/org/compiere/grid/ed/VLookup.java]: i assume that's the only real change from this class

File: de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java (hunk 1 above)
Comment (metas-rc) [inline de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java]: Yes.

File: de.metas.adempiere.adempiere/client/src/main/java-legacy/org/compiere/grid/ed/VLookup.java (hunk 5 above)
Comment (metas-rc) [inline de.metas.adempiere.adempiere/client/src/main/java-legacy/org/compiere/grid/ed/VLookup.java]: Right. This is the only change from VLookup, the rest is formatting.

File: de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java (hunk 3 above)
Comment (metas-rc) [inline de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java]: Fixed

File: de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java (hunk 2 above)
Comment (metas-rc) [inline de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessParametersPanel.java]: Fixed.

File: de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessPanel.java (hunk 4 above)
Comment (metas-rc) [inline de.metas.adempiere.adempiere/client/src/main/java/de/metas/process/ui/ProcessPanel.java]: Done

**Pull request merged**